### ISSUE TITLE
Add dev container definition for developing VS Code

### ISSUE THREAD

**Issue by Chuxel** [microsoft/vscode-dev-containers#339]
We would like to demonstrate the development of applications with a GUI component on Linux using VS Code as the example.

Current working draft is here: https://github.com/microsoft/vscode/tree/clantz/codespace

Goals:
1. Ensure this works on all platforms - macOS, Windows, Linux.
2. Ensure minimal setup - techniques like X forwarding are problematic for this reason.
3. Enable use in Codespaces over time using noVNC - this is blocked by https://github.com/MicrosoftDocs/vsonline/issues/117 currently, but we can get it in place for once it is available.

Non-goals: Provide a full desktop environment (e.g. XFCE). The goal would be to provide a way to start up VS Code and use VNC (or the web via noVNC) to see the GUI in a minimal desktop environment.

**Comment by PavelSosin-320** [comment]
Desktop GUI vs Browser GUI, what is better? I vote for browser GUI like Eclipse, SAP dev tools, and Github Sattelite. It is faster, more secure, and requires zero IT knowledge from the user. The application's user will never install something on his desktop. It is also Microsoft's strategy today. The developer must eat the same hay.

**Comment by bhack** [comment]
@Chuxel Have you explored also xpra html5 client https://xpra.org/trac/wiki/Clients/HTML5?

**Comment by PavelSosin-320** [comment]
@bhack, The additional Web browser-based UI has the test only if it is compatible with other compatible or potentially compatible solutions like VSCode, Eclipse Che, and SAP WeIDE. Otherwise, the solution will die from extension starvation.

**Comment by Chuxel** [comment]
@bhack That's an interesting thought -  I stuck with VNC given its ubiquity and that it is easy to have multiple clients connecting to a single desktop session. I actually found I got decent performance from VNC with a proper VNC client (not the Mac Screen Sharing app) which is exposed directly on 5901. noVNC also seems to provides a reasonable experience which is exposed on 6080. I'd have to see if I can easily get both up and running.  I'll fork that off as a separate investigation issue.
